docling_core/experimental/serializer/html.py

```diff
@@ -0,0 +1,922 @@
+
+                    html_parts.append("</td>")
+
+                    html_parts.append("<td>")
+                    html_parts.append(page.text)
+                    html_parts.append("</td>")
+
+                    html_parts.append("</tr>")
+                else:
+                    raise ValueError(
+                        "We need page-indices to leverage `split_page_view`"
+                    )
+
+            html_parts.append("</tbody>")
+            html_parts.append("</table>")
+
+        else:
+            # Add all pages
+            for page_no, page in pages.items():
+                if page.text:
+                    html_parts.append(page.text)
+
+        # Close HTML structure
+        html_parts.extend(["</body>", "</html>"])
+
+        # Join with newlines
+        html_content = "\n".join(html_parts)
+
+        return SerializationResult(text=html_content)
+
+    @override
+    def serialize_captions(
+        self,
+        item: FloatingItem,
+        tag: str = "figcaption",
+        **kwargs,
+    ) -> SerializationResult:
+        """Serialize the item's captions."""
+        caption_parts = []
+
+        # Extract caption text from all caption items
+        for cap in item.captions:
+            caption_item = cap.resolve(self.doc)
+            if isinstance(caption_item, TextItem):
+                caption_parts.append(caption_item.text)
+
+        # Join all captions with a space
+        if len(caption_parts) > 0:
+            caption_text = " ".join(caption_parts)
+            text_dir = get_text_direction(caption_text)
+
+            # Create proper HTML
+            if text_dir == "rtl":
+                return SerializationResult(
+                    text=f'<{tag} dir="{text_dir}">{html.escape(caption_text)}</{tag}>'
+                )
+            else:
+                return SerializationResult(
+                    text=f"<{tag}>{html.escape(caption_text)}</{tag}>"
+                )
+
+        return SerializationResult(text="")
+
+    def _generate_head(self) -> str:
+        """Generate the HTML head section with metadata and styles."""
+        params = self.params
+
+        head_parts = ["<head>", '<meta charset="UTF-8">']
+
+        # Add metadata if requested
+        if params.add_document_metadata:
+            if self.doc.name:
+                head_parts.append(f"<title>{html.escape(self.doc.name)}</title>")
+            else:
+                head_parts.append("<title>Docling Document</title>")
+
+            head_parts.append(
+                '<meta name="generator" content="Docling HTML Serializer">'
+            )
+
+        # Add default styles or custom CSS
+        if params.css_styles:
+            head_parts.append(f"<style>\n{params.css_styles}\n</style>")
+        elif self.params.split_page_view:
+            head_parts.append(_get_css_for_split_page())
+        else:
+            head_parts.append(_get_css_for_single_column())
```

Comment:
To make the result predictable, this code should check the that `output_style == OutputStyle.CUSTOM` (see suggestion above), and raise an exception when `css_styles` is provided but the output_style is not CUSTOM.